### ISSUE TITLE
Object has no method 'unset'

### ISSUE THREAD

**Issue by dtodt** [aldeed/meteor-simple-schema#562]
Greetings.

I'm tring to configure a simple-schema on my project with auto values.
But when I do Collection.insert, I get the exception bellow:
```
C:\Users\Daniel\AppData\Local\.meteor\packages\meteor-tool\1.1.13-beta.11\mt-os.windows.x86_32\dev_bundle\server-lib\node_modules\fibers\future.js:245
                                           throw(ex);
                                                 ^
TypeError: Object [object global] has no method 'unset'
    at Object.runAV (packages/aldeed_simple-schema/simple-schema.js:310:1)
    at packages/aldeed_simple-schema/simple-schema.js:411:1
    at Array.forEach (packages/es5-shim/.npm/package/node_modules/es5-shim/es5-shim.js:419:1)
    at Function._.each._.forEach (packages/underscore/underscore.js:105:1)
    at Object.autoValue (lib/collections/providers.js:19:22)
    at packages/aldeed_simple-schema/simple-schema.js:410:1
    at Function._.each._.forEach (packages/underscore/underscore.js:113:1)
    at [object Object].getAutoValues (packages/aldeed_simple-schema/simple-schema.js:373:1)
    at [object Object].SimpleSchema.clean (packages/aldeed_simple-schema/simple-schema.js:791:1)
    at doClean (packages/aldeed_collection2-core/lib/collection2.js:279:1)
```

I'm not sure if is something that I did wrong or maybe something from meteor wich is in beta.

*My setup:*
S.O: Windows 10
Node: v5.7.0
Npm: 3.7.2
Meteor: 1.3-beta.11
aldeed:collection2: 2.9.0
aldeed:collection2-core: 1.1.0
aldeed:schema-deny: 1.0.1
aldeed:schema-index: 1.0.1
aldeed:simple-schema: 1.5.3

My collection definition:
```
import {Mongo} from 'meteor/mongo';
import {SimpleSchema} from 'meteor/aldeed:simple-schema';

const providers = new Mongo.Collection('providers');

let schema = new SimpleSchema({
  name: {
    type: String
  },
  createdAt: {
    type: Date,
    autoValue: () => {
      let value;
      if (this.isInsert) {
        value = new Date();
      } else if (this.isUpsert) {
        value = {$setOnInsert: new Date()};
      } else {
        this.unset();
      }
      return value;
    }
  },
  updatedAt: {
    type: Date,
    autoValue: () => {
      let value;
      if (this.isUpdate) {
        value = new Date();
      }
      return value;
    },
    denyInsert: true,
    optional: true
  }
});

providers.attachSchema(schema);

export default providers;
```

**Comment by rub1e** [comment]
**tl;dr**
If you haven't already moved on, it's a simple fix - just call `function(){}` instead of `() => {}`

@dtodt just been struggling through the same problem - your issue is the lexical `this` in the arrow functions you're using isn't what you think it is - see [here](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Lexical_this)
> Arrow functions capture the this value of the enclosing context

For my edification, can anyone tell me how to do this without changing the arrow function?

**Comment by brewhk-dev** [comment]
The whole point of the arrow function is to provide a shorter syntax 

So instead of this:

```
var self = this;
self.methodWithCallback(arg, function() {
    self.doSomething();
});
```

You have this:

```
this.methodWithCallback(arg, () => {
    this.doSomething();
});
```

If you don't need this binding, just use the normal `function () {}`

**Comment by lonniev** [comment]
The moral of this painful story is: if you see traditional `function() {...}` use in someone else's code, do not modernize it without first checking if its body uses `this`. If it does, leave it old-style.
